### PR TITLE
Release 3.2.0

### PR DESCRIPTION
**Pull request by nicoddemus**
Builds:

* [Build status](https://travis-ci.org/nicoddemus/devpi-cloud-test-pytest)

* [Build Status](https://ci.appveyor.com/project/nicoddemus/devpi-cloud-test-pytest)

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -8,6 +8,80 @@
...
+
+Deprecations and Removals
+-------------------------
+
+- All old-style specific behavior in current classes in the pytest's API is
+  considered deprecated at this point and will be removed in a future release.
+  This affects Python 2 users only and in rare situations. (`#2147
+  <https://github.com/pytest-dev/pytest/issues/2147>`_)
+
+- A deprecation warning is now raised when using marks directly into parameters
```
Comment (The-Compiler) [inline CHANGELOG.rst]: into parameters -> for parameters / with parameters?

File: CHANGELOG.rst [hunk 2]
```diff
@@ -8,6 +8,80 @@
...
+- Collection ignores local virtualenvs by default; `--collect-in-virtualenv`
+  overrides this behavior. (`#2518 <https://github.com/pytest-
+  dev/pytest/issues/2518>`_)
+
+- Allow class methods decorated as ``@staticmethod`` to be candidates for
+  collection as a test function. (Only for Python 2.7 and above. Python 2.6
+  will still ignore static methods.) (`#2528 <https://github.com/pytest-
+  dev/pytest/issues/2528>`_)
+
+- New ``cache_dir`` ini option: sets a directory where stores content of cache
```
Comment (The-Compiler) [inline CHANGELOG.rst]: where content of the cache plugin is stored?

File: CHANGELOG.rst (hunk 1 above)
Comment (nicoddemus) [inline CHANGELOG.rst]: Yep, that's better.

We should also add an example to the docs or here if possible on how to update the code, as mentioned by @fschulze in the ML.

File: CHANGELOG.rst (hunk 2 above)
Comment (nicoddemus) [inline CHANGELOG.rst]: This probably would be better written as: "sets the directory where the contents of the cache plugin are stored."

**Comment by The-Compiler** [comment]
#2573 sounds severe enough to stop this, I think.

**Comment by nicoddemus** [comment]
> sounds severe enough to stop this, I think.

I agree, just added a comment there. We should definitely fix the error handling pointed out there and improve the docs about the deprecated feature.

**Comment by nicoddemus** [comment]
@The-Compiler could you please mark this PR as "Changes requested" to make it clear we have more work to do for the release? I would like to keep all in the same PR so we can track comments and change requests.

**Comment by nicoddemus** [comment]
(I would but I can't since I'm the author of the PR)

**Comment by nicoddemus** [comment]
Closing this for new because it will probably be more work to merge further things here. I will open a new PR when the time is right for the 3.2 release.

**Pull request closed without merging**